core/src/main/java/org/lflang/generator/c/CCmakeGenerator.java

```diff
@@ -247,12 +327,13 @@ CodeBuilder generateCMakeCode(
     }
 
     if (targetConfig.threading || targetConfig.tracing != null) {
-      // If threaded computation is requested, add the threads option.
-      cMakeCode.pr("# Find threads and link to it");
-      cMakeCode.pr("find_package(Threads REQUIRED)");
-      cMakeCode.pr("target_link_libraries(${LF_MAIN_TARGET} PRIVATE Threads::Threads)");
-      cMakeCode.newLine();
-
+      if (targetConfig.platformOptions.platform != Platform.RP2040) {
```

Comment:
I think a check like this should be done earlier, during validation. Specifically, if the platform is `RP2040`, then `threading` must not be `true` (or an error should result).

Reply:
I've pushed a fix, which works, but I'm not very happy about how it works. I'll try to push improvements at a later time...